lib/utils/index.ts

```diff
@@ -260,7 +259,7 @@ export function parseURL(url) {
     options.allowUsernameInURI && options.allowUsernameInURI !== "false";
   delete options.allowUsernameInURI;
 
-  const result: any = {};
+  const result: any = { ...options };
```

Comment:
This is not equivalent. Better do an Object.assign in line 287 instead of removing it.